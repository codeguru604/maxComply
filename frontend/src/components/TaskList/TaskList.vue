<template>
	<div
		className="list-container"
	>
		<q-toolbar>
			<q-toolbar-title>Task List</q-toolbar-title>
		</q-toolbar>
		<q-input
			v-model="filterData"
			outlined
			label="Search List..."
			class="list-search"
			@keyup="changeFilter"
		>
			<template #prepend>
				<q-icon name="search" />
			</template>
		</q-input>
		<q-list
			bordered
			separator
		>
			<q-item
				v-for="task in listData"
				:key="task.id"
				v-ripple
				outlined
				clickable
				class="list-item"
				@click="selectTask(task.id)"
			>
				<q-item-section>
					<q-item-label class="list-item-title">
						{{ task.name }}
					</q-item-label>
					<q-item-label caption>
						{{ task.subHeading }}
					</q-item-label>
				</q-item-section>
			</q-item>
		</q-list>
	</div>
</template>

<script setup>
import { ref, onMounted } from 'vue';
import { useTaskStore } from '../../stores/taskStore';
import { Notify } from 'quasar';
const loading = ref(false);
const taskStore = useTaskStore();
const filterData = ref('');
const listData = ref([]);

const showNotify = (message, color, loading) => {
	Notify.create({
		spinner: loading,
		message: message,
		color: color,
		position: 'top',
		timeout: 2000,
	});
};
const changeFilter = () => {
	if (filterData.value) {
		listData.value = taskStore.tasks.filter(task => {
			return task.name.toUpperCase().includes(filterData.value.toUpperCase());
		});
	} else {
		listData.value = taskStore.tasks;
	}
};
const selectTask = (taskId) => {
	loading.value = true;
	showNotify('Loading', 'blue', true);
	taskStore.fetchTaskById(taskId).then(() => {
		loading.value = false;
	});
};

onMounted(() => {
	loading.value = true;
	taskStore.fetchTasks().then(() => {
		if (taskStore.tasks.length > 0) {
			listData.value = taskStore.tasks;
			changeFilter();
			selectTask(taskStore.tasks[0].id);
		}
		loading.value = false;
	});
});
</script>

<style scope lang="scss">
@import '../../css/app.scss';
</style>