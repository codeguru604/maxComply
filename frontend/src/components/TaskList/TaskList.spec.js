import { mount } from '@vue/test-utils';
import TaskList from './TaskList.vue';
import { createPinia, setActivePinia } from 'pinia';

const mockFetchTasks = jest.fn(() => Promise.resolve([
	{
		id: 1,
		name: 'Task 1',
		subHeading: 'Subtask 1',
	}
]));
const mockFetchTaskById = jest.fn(() => Promise.resolve());

jest.mock('../../stores/taskStore', () => ({
	useTaskStore: () => ({
		tasks: [
			{
				id: 1,
				name: 'Task 1',
				subHeading: 'Subtask 1',
			}
		],
		fetchTasks: mockFetchTasks,
		fetchTaskById: mockFetchTaskById,
	}),
}));

jest.mock('quasar', () => ({
	Notify: {
		create: jest.fn(),
		setDefaults: jest.fn(),
		registerType: jest.fn(),
	},
}));

describe('TaskList', () => {
	beforeEach(() => {
		const pinia = createPinia();
		setActivePinia(pinia);
	});

	it('renders a task list', async () => {
		const wrapper = mount(TaskList);
		await wrapper.vm.$nextTick();
		await wrapper.vm.$nextTick();
		expect(wrapper.find('.list-item').exists()).toBe(true);
		expect(wrapper.element).toMatchSnapshot();
	});
});
