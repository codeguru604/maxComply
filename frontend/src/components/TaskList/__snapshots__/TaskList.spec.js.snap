// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList renders a task list 1`] = `
<div
  class="list-container"
>
  <q-toolbar>
    <q-toolbar-title>
      Task List
    </q-toolbar-title>
  </q-toolbar>
  <q-input
    class="list-search"
    label="Search List..."
    modelvalue=""
    outlined=""
  />
  <q-list
    bordered=""
    separator=""
  >
    
    <q-item
      class="list-item"
      clickable=""
      outlined=""
    >
      <q-item-section>
        <q-item-label
          class="list-item-title"
        >
          Task 1
        </q-item-label>
        <q-item-label
          caption=""
        >
          Subtask 1
        </q-item-label>
      </q-item-section>
    </q-item>
    
  </q-list>
</div>
`;
